### DISCUSSION TITLE
Using the `removed` block for resource migration?

### DISCUSSION THREAD

**Discussion by paulhalla** [Snowflake-Labs/terraform-provider-snowflake#2944] [Q&A] [answered]
[Starting from 1.7](https://developer.hashicorp.com/terraform/language/resources/syntax#removing-resources), you can use the `removed` block to remove a resource from the Terraform state without destroying the actual infrastructure object. 

When migrating deprecated resource types to new ones, would not using the `removed` block be a better alternative than using `terraform state rm <resource_address>`? Just wanted to sense check if there's a reason the [resource migration documentation](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/docs/technical-documentation/resource_migration.md) uses the latter approach instead of using the removed block functionality?

**Comment by sfc-gh-asawicki** [comment, marked as answer]
Hey @paulhalla. The resource migration doc was written some time ago. We would have to verify the usage of these blocks before we post it officially, but they should work.

Also, not every user has 1.7.x, and sometimes, we have to encourage users to migrate to 1.1.5, so having a way to work also in older Terraform versions seems like a good idea.

We may include the `removed` block as an alternative for the newer versions. We will also accept a contribution with a step-by-step guide for this alternative (following the https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/CONTRIBUTING.md).
